src/elisp/treemacs-impl.el

```diff
@@ -773,14 +778,20 @@ failed."
             (goto-char start))
         search-result))))
 
-(define-inline treemacs--goto-custom-top-level-node (path)
-  (inline-letevals (path)
-    (inline-quote
-     (-when-let (result (treemacs--find-custom-top-level-node ,path))
-       (treemacs--evade-image)
-       (hl-line-highlight)
-       (set-window-point (get-buffer-window) (point))
-       result))))
+(defun treemacs--find-custom-project-node (path)
+  "Move to the project extension node at PATH."
+  (unless (treemacs-project->is-expanded? (car path))
+    (treemacs--expand-root-node (car path)))
+  (or (-some-> (treemacs-find-in-dom path)
+               (treemacs-dom-node->position))
+      (let* ((project (car path))
+             (start (point))
+             (project-start (treemacs-project->position project))
+             (project-end (progn (goto-char project-start)
+                                 (treemacs--next-project-pos)))
+             (pos (text-property-any project-start project-end :key (nth 1 path))))
```

Comment:
And this is where the bug is. I somehow thought that the path consisted of project and key, not of project followed by path elements.

Reply:
Also, I think there's a second bug here, as I think that searching for text properties requires the search key to be eq-comparable.